python/pyspark/ml/tuning.py

```diff
@@ -261,19 +260,30 @@ def setParams(self, estimator=None, estimatorParamMaps=None, evaluator=None, num
         kwargs = self._input_kwargs
         return self._set(**kwargs)
 
-    @since("1.4.0")
-    def setNumFolds(self, value):
+    def setEstimator(self, value):
         """
-        Sets the value of :py:attr:`numFolds`.
+        Sets the value of :py:attr:`estimator`.
         """
-        return self._set(numFolds=value)
+        return self._set(estimator=value)
+
+    def setEstimatorParamMaps(self, value):
+        """
+        Sets the value of :py:attr:`estimatorParamMaps`.
+        """
+        return self._set(estimatorParamMaps=value)
+
+    def setEvaluator(self, value):
```

Comment:
Do we need to add since for setters? Some like setNumFolds have it, but some do not.

Reply:
If the original file has @since, I have it too. Some of the getters/setters don't have @since in the original file, so I don't have i too.
I will check the history to find out which release starts to have these getters/setters, and add @since for each of the getters/setters.

Reply:
I checked the history, the ```@since``` was removed in https://github.com/apache/spark/pull/12020. I guess the reason is that in ```CrossValidator```, the getters/getters have ```@since 1.4```, but in ```TrainValidationSplit```, the getters/getters have ```@since 2.0```. When putting the common getters/setters in ```ValidatorParams```, neither ```@since 1.4``` nor ```@since 2.0``` is quite right, so ```@since``` was removed. I am facing the same problem. Shall I have ```@since 2.0```? or just not adding back the ```@since```?

Reply:
`@since 2.0` is at least accurate, as both have existed since that version. One existed before, sure, but this is maybe not a big deal, as it won't matter much at all to know it was in 1.x.